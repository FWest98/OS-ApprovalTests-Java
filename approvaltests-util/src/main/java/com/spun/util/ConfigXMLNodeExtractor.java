package com.spun.util;

import com.spun.util.io.XMLNodeExtractor;
import org.w3c.dom.Node;
import org.w3c.dom.NodeList;

import java.util.ArrayList;
import java.util.HashMap;

public class ConfigXMLNodeExtractor implements XMLNodeExtractor
{
  public void extractProperty(Node node, HashMap<String, Object> properties)
  {
    //String name = node.getNodeName();
    if (XMLNodeExtractor.Utils.extractSingleton(node, properties))
    {
      // do nothing
    }
    else if (node.getChildNodes().getLength() > 0)
    {
      NodeList childNodes = node.getChildNodes();
      ArrayList<String> v = new ArrayList<String>();
      for (int i = 0; i < childNodes.getLength(); i++)
      {
        if (childNodes.item(i).getNodeName().equals("VALUE"))
        {
          v.add(StringUtils.loadNullableString(childNodes.item(i).getChildNodes().item(0).getNodeValue()));
        }
      }
      properties.put(node.getNodeName(), StringUtils.toArray(v));
    }
  }
}
