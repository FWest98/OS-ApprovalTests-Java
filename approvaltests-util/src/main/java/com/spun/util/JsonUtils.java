package com.spun.util;

import com.google.gson.*;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;
import org.lambda.functions.Function1;

import java.io.IOException;
import java.time.Instant;

public class JsonUtils
{
  public static String prettyPrint(String json)
  {
    if (!ObjectUtils.isClassPresent("com.google.gson.Gson"))
    {
      throw new RuntimeException(
          "Missing Gson dependency\n  Pretty print uses Gson parser.\n  You can get this from the maven repo \n  or https://github.com/google/gson");
    }
    try
    {
      Gson gson = new GsonBuilder().setPrettyPrinting().create();
      JsonElement je = JsonParser.parseString(json);
      return gson.toJson(je);
    }
    catch (JsonSyntaxException e)
    {
      return String.format("Error:%s\nJson:\n%s", e.getMessage(), json);
    }
  }
  public static String asJson(Object o)
  {
    return asJson(o, (b) -> b);
  }
  public static <T> String asJsonWithBuilder(Object o, Function1<T, T> gsonBuilder, Class<T> gsonBuilderClass)
  {
    if (!ObjectUtils.isThisInstanceOfThat(gsonBuilderClass, com.google.gson.GsonBuilder.class))
    { throw new FormattedException("Class must be of type %s", GsonBuilder.class.getName()); }
    return asJson(o, (Function1<GsonBuilder, GsonBuilder>) gsonBuilder);
  }
  public static <T> String asJson(Object o, Function1<GsonBuilder, GsonBuilder> gsonBuilder)
  {
    GsonBuilder builder = new GsonBuilder().setPrettyPrinting();
    builder = addHandlingForDateObjects(builder);
    Gson gson = gsonBuilder.call(builder).create();
    return gson.toJson(o);
  }
  private static GsonBuilder addHandlingForDateObjects(GsonBuilder builder)
  {
    builder = builder.registerTypeAdapter(Instant.class, new InstantAdapter());
    return builder;
  }
  public static class InstantAdapter extends TypeAdapter<Instant>
  {
    @Override
    public void write(JsonWriter jsonWriter, Instant instant) throws IOException
    {
      jsonWriter.value(instant.toString());
    }
    @Override
    public Instant read(JsonReader jsonReader) throws IOException
    {
      throw new IOException("Never called");
    }
  }
}
