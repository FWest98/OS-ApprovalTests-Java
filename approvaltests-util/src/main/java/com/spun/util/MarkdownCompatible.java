package com.spun.util;

public interface MarkdownCompatible
{
  String toMarkdown();
}
