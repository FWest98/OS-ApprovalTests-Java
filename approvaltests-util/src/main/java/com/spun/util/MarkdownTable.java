package com.spun.util;

import org.lambda.functions.Function1;
import org.lambda.query.Queryable;

public class MarkdownTable implements MarkdownCompatible
{
  protected String markdown;
  public static <I, O> MarkdownTable create(I[] inputs, Function1<I, O> o, String column1, String column2)
  {
    MarkdownTable table = new MarkdownTable().withColumnHeaders(column1, column2);
    for (I input : inputs)
    {
      table.addRow(input, o.call(input));
    }
    return table;
  }
  public static MarkdownTable withHeaders(String... columnNames)
  {
    MarkdownTable table = new MarkdownTable();
    return table.withColumnHeaders(columnNames);
  }
  public <I> MarkdownTable addRowsForInputs(I[] inputs, Function1<I, Object>... transfers)
  {
    for (I input : inputs)
    {
      Queryable<Object> row = Queryable.as(transfers).select(f -> f.call(input));
      row.add(0, input);
      addRow(row.toArray());
    }
    return this;
  }
  public MarkdownTable addRow(Object... columns)
  {
    markdown += printRow(columns);
    return this;
  }
  public MarkdownTable withColumnHeaders(String... headers)
  {
    markdown = printColumnHeaders(headers);
    return this;
  }
  @Override
  public String toString()
  {
    return toMarkdown();
  }
  @Override
  public String toMarkdown()
  {
    return markdown;
  }
  public static String printColumnHeaders(String... headers)
  {
    return printRow(headers) + printRow(ArrayUtils.of("---", headers.length));
  }
  public static String printRow(Object... columns)
  {
    StringBuffer b = new StringBuffer();
    b.append("|");
    for (int x = 0; x < columns.length; ++x)
    {
      b.append(String.format(" %s |", columns[x]));
    }
    b.append("\n");
    return b.toString();
  }
}
