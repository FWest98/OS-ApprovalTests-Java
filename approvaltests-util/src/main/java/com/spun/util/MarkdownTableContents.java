package com.spun.util;

public class MarkdownTableContents implements MarkdownTableElement, Resizable
{
  private String contents;
  private int    padUntil;
  public MarkdownTableContents(String contents)
  {
    this.contents = contents;
  }
  @Override
  public String toString()
  {
    return " " + StringUtils.pad(contents, padUntil) + " ";
  }
  public int getLength()
  {
    return contents.length();
  }
  public void setPadding(int length)
  {
    padUntil = length;
  }
}
