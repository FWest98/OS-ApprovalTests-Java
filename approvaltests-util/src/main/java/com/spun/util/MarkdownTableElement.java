package com.spun.util;

public interface MarkdownTableElement {
    MarkdownTableElement DELIMITER = new MarkdownTableElement() {
        @Override
        public String toString() {
            return "|";
        }
    };
    MarkdownTableElement NEWLINE = new MarkdownTableElement() {
        @Override
        public String toString() {
            return "\n";
        }
    };
}
