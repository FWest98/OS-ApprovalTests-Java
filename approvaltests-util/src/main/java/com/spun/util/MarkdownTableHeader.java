package com.spun.util;

public class MarkdownTableHeader implements MarkdownTableElement, Resizable
{
  private int padUntil = 1;
  public MarkdownTableHeader()
  {
  }
  @Override
  public String toString()
  {
    int dashes = padUntil + 2; // Math.max(3, padUntil + 2);
    return "----------------------------------".substring(0, dashes);
  }
  public int getLength()
  {
    return 1;
  }
  public void setPadding(int length)
  {
    padUntil = length;
  }
}
