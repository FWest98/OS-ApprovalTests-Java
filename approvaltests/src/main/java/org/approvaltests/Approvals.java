package org.approvaltests;

import com.spun.util.*;
import com.spun.util.persistence.ExecutableQuery;
import com.spun.util.persistence.Loader;
import com.spun.util.persistence.SqlLoader;
import org.approvaltests.approvers.ApprovalApprover;
import org.approvaltests.approvers.FileApprover;
import org.approvaltests.core.ApprovalFailureReporter;
import org.approvaltests.core.ApprovalWriter;
import org.approvaltests.namer.ApprovalNamer;
import org.approvaltests.namer.StackTraceNamer;
import org.approvaltests.reporters.ExecutableQueryFailure;
import org.approvaltests.writers.*;
import org.lambda.actions.Action0;
import org.lambda.functions.Function1;
import org.lambda.query.Query;

import java.io.File;
import java.io.FileFilter;
import java.io.FilenameFilter;
import java.sql.ResultSet;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.Map;

public class Approvals
{
  public static Loader<ApprovalNamer> namerCreater = new Loader<ApprovalNamer>()
  {
    public ApprovalNamer load()
    {
      return new StackTraceNamer();
    }
  };
  public static void verify(String response)
  {
    verify(new ApprovalTextWriter(response, "txt"));
  }
  public static void verify(Object o)
  {
    verify("" + o);
  }
  public static <T> void verifyAll(String label, T[] array)
  {
    verify(new ApprovalTextWriter(StringUtils.toString(label, array), "txt"));
  }
  public static <T> void verifyAll(String header, String label, T[] array)
  {
    verify(new ApprovalTextWriter(formatHeader(header) + StringUtils.toString(label, array), "txt"));
  }
  public static <T> void verifyAll(T[] values, Function1<T, String> f1)
  {
    String text = ArrayUtils.toString(values, f1);
    verify(new ApprovalTextWriter(text, "txt"));
  }
  public static <T> void verifyAll(String header, T[] values, Function1<T, String> f1)
  {
    verifyAll(header, Arrays.asList(values), f1);
  }
  public static <T> void verifyAll(String header, Iterable<T> array, Function1<T, String> f1)
  {
    String text = formatHeader(header) + ArrayUtils.toString(array, f1);
    verify(new ApprovalTextWriter(text, "txt"));
  }
  private static String formatHeader(String header)
  {
    return StringUtils.isEmpty(header) ? "" : header + "\n\n\n";
  }
  public static <T> void verifyAll(String label, Iterable<T> array)
  {
    verify(new ApprovalTextWriter(StringUtils.toString(label, array), "txt"));
  }
  public static <T> void verifyAll(String header, String label, Iterable<T> array)
  {
    verify(new ApprovalTextWriter(formatHeader(header) + StringUtils.toString(label, array), "txt"));
  }

  public static void verifyHtml(String response)
  {
    verify(new ApprovalTextWriter(response, "html"));
  }
  public static void verify(File generateFile)
  {
    verify(new FileApprovalWriter(generateFile));
  }

  public static void verify(ApprovalWriter writer, ApprovalNamer namer, ApprovalFailureReporter reporter)
  {
    verify(new FileApprover(writer, namer), reporter);
  }
  public static void verify(ApprovalWriter writer)
  {
    verify(writer, createApprovalNamer(), getReporter());
  }
  public static void verifyXml(String xml)
  {
    verify(new ApprovalXmlWriter(xml));
  }
  public static void verify(ApprovalApprover approver)
  {
    verify(approver, getReporter());
  }
  public static void verify(ApprovalApprover approver, ApprovalFailureReporter reporter)
  {
    if (!approver.approve())
    {
      approver.reportFailure(reporter);
      approver.fail();
    }
    else
    {
      approver.cleanUpAfterSuccess(reporter);
    }
  }
  public static void verify(ExecutableQuery query)
  {
    verify(new ApprovalTextWriter(query.getQuery(), "txt"), createApprovalNamer(),
        new ExecutableQueryFailure(query));
  }
  public static void verify(Map<?, ?> map)
  {
    verify(new ApprovalTextWriter(StringUtils.toString(map), "txt"));
  }
  public static void verify(ResultSet rs)
  {
    verify(new ResultSetApprovalWriter(rs));
  }
  public static <T> void verify(SqlLoader<T> loader)
  {
    verify(new SqlLoader.ExecutableWrapper<T>(loader));
  }
  public static ApprovalNamer createApprovalNamer()
  {
    return namerCreater.load();
  }

  public static void verifyEachFileInDirectory(File directory)
  {
    verifyEachFileAgainstMasterDirectory(directory.listFiles());
  }
  public static void verifyEachFileInDirectory(File directory, FileFilter filter)
  {
    verifyEachFileAgainstMasterDirectory(directory.listFiles(filter));
  }
  public static void verifyEachFileInDirectory(File directory, FilenameFilter filter)
  {
    verifyEachFileAgainstMasterDirectory(directory.listFiles(filter));
  }
  private static void verifyEachFileAgainstMasterDirectory(File[] files) throws Error
  {
    ApprovalNamer namer = createApprovalNamer();
    String dirName = namer.getSourceFilePath() + File.separator + namer.getApprovalName() + ".Files";
    File approvedDirectory = new File(dirName);
    List<File> mismatched = new ArrayList<File>();
    for (File f : files)
    {
      if (!f.isDirectory())
      {
        try
        {
          verify(new DirectoryToDirectoryWriter(f, approvedDirectory));
        }
        catch (Throwable e)
        {
          mismatched.add(f);
        }
      }
    }
    if (!mismatched.isEmpty())
    {
      String message = "The Following Files did not match up: " + getFileNameList(mismatched);
      throw new Error(message);
    }
  }
  private static String getFileNameList(List<File> mismatched)
  {
    return Query.select(mismatched, a -> a.getName()).toString();
  }
  public static void verifyJson(String json)
  {
    verify(JsonUtils.prettyPrint(json), "json");
  }
  public static void verify(Object actual, String fileExtensionWithoutDot)
  {
    verify(new ApprovalTextWriter("" + actual, fileExtensionWithoutDot));
  }
  public static ApprovalFailureReporter getReporter()
  {
    return ReporterFactory.get();
  }
  public static void verifyAsJson(Object o)
  {
    verify(JsonUtils.asJson(o), "json");
  }
  public static void verifyException(Action0 runnableBlock)
  {
    Throwable t = ObjectUtils.captureException(runnableBlock);
    if (t == null) { throw new FormattedException("No exception thrown when running %s", runnableBlock); }
    Approvals.verify(String.format("%s: %s", t.getClass().getName(), t.getMessage()));
  }
}