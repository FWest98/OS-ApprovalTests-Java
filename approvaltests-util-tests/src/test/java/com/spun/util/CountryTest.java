package com.spun.util;

import org.approvaltests.Approvals;
import org.junit.Test;

import com.spun.util.Country;

public class CountryTest
{
  @Test
  public void allCountries()
  {
    Approvals.verifyAll("Countries", Country.getStringValues());
  }
}
