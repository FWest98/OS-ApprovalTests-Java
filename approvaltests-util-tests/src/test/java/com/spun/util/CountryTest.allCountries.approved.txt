Countries[000] = Albania
Countries[001] = Algeria
Countries[002] = American Samoa
Countries[003] = Andorra
Countries[004] = Angola
Countries[005] = Anguilla
Countries[006] = Antigua & Barbuda
Countries[007] = Argentina
Countries[008] = Armenia
Countries[009] = Aruba
Countries[010] = Australia
Countries[011] = Austria
Countries[012] = Azerbaijan
Countries[013] = Azores
Countries[014] = Bahamas
Countries[015] = Bahrain
Countries[016] = Bangladesh
Countries[017] = Barbados
Countries[018] = Belarus
Countries[019] = Belgium
Countries[020] = Belize
Countries[021] = Benin
Countries[022] = Bermuda
Countries[023] = Bolivia
Countries[024] = Bonaire
Countries[025] = Bosnia
Countries[026] = Botswana
Countries[027] = Brazil
Countries[028] = British Virgin Isles
Countries[029] = Brunei
Countries[030] = Bulgaria
Countries[031] = Burkina Faso
Countries[032] = Burundi
Countries[033] = Cambodia
Countries[034] = Cameroon
Countries[035] = Canada
Countries[036] = Canary Islands
Countries[037] = Cape Verde
Countries[038] = Cayman Islands
Countries[039] = Central African Republic
Countries[040] = Chad
Countries[041] = Channel Islands
Countries[042] = Chile
Countries[043] = Colombia
Countries[044] = Cook Islands
Countries[045] = Costa Rica
Countries[046] = Cote d' Ivoire
Countries[047] = Croatia
Countries[048] = Cuba
Countries[049] = Curacao
Countries[050] = Cyprus
Countries[051] = Czech Republic
Countries[052] = Dem. Republic of Congo
Countries[053] = Denmark
Countries[054] = Djibouti
Countries[055] = Dominica
Countries[056] = Dominican Republic
Countries[057] = Ecuador
Countries[058] = Egypt
Countries[059] = El Salvador
Countries[060] = England
Countries[061] = Equatorial Guinea
Countries[062] = Eritrea
Countries[063] = Estonia
Countries[064] = Ethiopia
Countries[065] = Faeroe Islands
Countries[066] = Fiji
Countries[067] = Finland
Countries[068] = France
Countries[069] = French Guiana
Countries[070] = French Polynesia
Countries[071] = Gabon
Countries[072] = Gambia
Countries[073] = Georgia
Countries[074] = Germany
Countries[075] = Ghana
Countries[076] = Gibraltar
Countries[077] = Greece
Countries[078] = Greenland
Countries[079] = Grenada
Countries[080] = Guadeloupe
Countries[081] = Guam
Countries[082] = Guatemala
Countries[083] = Guinea
Countries[084] = Guinea-Bissau
Countries[085] = Guyana
Countries[086] = Haiti
Countries[087] = Holland
Countries[088] = Honduras
Countries[089] = Hong Kong
Countries[090] = Hungary
Countries[091] = Iceland
Countries[092] = India
Countries[093] = Indonesia
Countries[094] = Iran
Countries[095] = Iraq
Countries[096] = Israel
Countries[097] = Italy
Countries[098] = Jamaica
Countries[099] = Japan
Countries[100] = Jordan
Countries[101] = Kazakhstan
Countries[102] = Kenya
Countries[103] = Kiribati
Countries[104] = Kosrae
Countries[105] = Kuwait
Countries[106] = Kyrgyzstan
Countries[107] = Laos
Countries[108] = Latvia
Countries[109] = Lebanon
Countries[110] = Lesotho
Countries[111] = Liberia
Countries[112] = Libya
Countries[113] = Liechtenstein
Countries[114] = Lithuania
Countries[115] = Luxembourg
Countries[116] = Macau
Countries[117] = Macedonia
Countries[118] = Madagascar
Countries[119] = Madeira
Countries[120] = Malawi
Countries[121] = Malaysia
Countries[122] = Maldives
Countries[123] = Mali
Countries[124] = Malta
Countries[125] = Marshall Islands
Countries[126] = Martinique
Countries[127] = Mauritania
Countries[128] = Mauritius
Countries[129] = Mexico
Countries[130] = Micronesia
Countries[131] = Moldova
Countries[132] = Monaco
Countries[133] = Mongolia
Countries[134] = Montserrat
Countries[135] = Morocco
Countries[136] = Mozambique
Countries[137] = Myanmar
Countries[138] = N. Mariana Islands
Countries[139] = Namibia
Countries[140] = Nepal
Countries[141] = Netherlands
Countries[142] = Netherlands Antilles
Countries[143] = New Caledonia
Countries[144] = New Zealand
Countries[145] = Nicaragua
Countries[146] = Niger
Countries[147] = Nigeria
Countries[148] = Norfolk Island
Countries[149] = North Korea
Countries[150] = Northern Ireland
Countries[151] = Norway
Countries[152] = Oman
Countries[153] = Pakistan
Countries[154] = Palau
Countries[155] = Panama
Countries[156] = Papua New Guinea
Countries[157] = Paraguay
Countries[158] = Peoples Republic of China
Countries[159] = Peru
Countries[160] = Philippines
Countries[161] = Poland
Countries[162] = Ponape
Countries[163] = Portugal
Countries[164] = Puerto Rico
Countries[165] = Qatar
Countries[166] = Republic of Congo
Countries[167] = Republic of Ireland
Countries[168] = Republic of Yemen
Countries[169] = Reunion
Countries[170] = Romania
Countries[171] = Rota
Countries[172] = Russia
Countries[173] = Rwanda
Countries[174] = Saba
Countries[175] = Saipan
Countries[176] = San Marino
Countries[177] = Saudi Arabia
Countries[178] = Scotland
Countries[179] = Senegal
Countries[180] = Seychelles
Countries[181] = Sierra Leone
Countries[182] = Singapore
Countries[183] = Slovakia
Countries[184] = Slovenia
Countries[185] = Solomon Islands
Countries[186] = South Africa
Countries[187] = South Korea
Countries[188] = Spain
Countries[189] = Sri Lanka
Countries[190] = St Vincent/Grenadine
Countries[191] = St. Barthelemy
Countries[192] = St. Christopher
Countries[193] = St. Croix
Countries[194] = St. Eustatius
Countries[195] = St. John
Countries[196] = St. Kitts & Nevis
Countries[197] = St. Lucia
Countries[198] = St. Maarten
Countries[199] = St. Martin
Countries[200] = St. Thomas
Countries[201] = Sudan
Countries[202] = Suriname
Countries[203] = Swaziland
Countries[204] = Sweden
Countries[205] = Switzerland
Countries[206] = Syria
Countries[207] = Tahiti
Countries[208] = Taiwan
Countries[209] = Tajikistan
Countries[210] = Tanzania
Countries[211] = Thailand
Countries[212] = Tinian
Countries[213] = Togo
Countries[214] = Tonga
Countries[215] = Tortola
Countries[216] = Trinidad & Tobago
Countries[217] = Truk
Countries[218] = Tunisia
Countries[219] = Turkey
Countries[220] = Turkmenistan
Countries[221] = Turks & Caicos Isles
Countries[222] = Tuvalu
Countries[223] = Uganda
Countries[224] = Ukraine
Countries[225] = Union Island
Countries[226] = United Arab Emirates
Countries[227] = United Kingdom
Countries[228] = United States
Countries[229] = Uruguay
Countries[230] = US Virgin Islands
Countries[231] = Uzbekistan
Countries[232] = Vanuatu
Countries[233] = Vatican City State
Countries[234] = Venezuela
Countries[235] = Vietnam
Countries[236] = Virgin Gorda
Countries[237] = Wake Island
Countries[238] = Wales
Countries[239] = Wallis & Futuna Isle
Countries[240] = Western Samoa
Countries[241] = Yap
Countries[242] = Yugoslavia
Countries[243] = Zambia
Countries[244] = Zimbabwe
